### PR TITLE
Make deterministic an option.

### PR DESCRIPTION
**Pull request by kevinzakka**
For debugging and overfitting.

### CODE REVIEW

**Comment by kpertsch** [comment]
Did you test whether this actually makes the data loader fully deterministic? For very large model training it can be good to have a deterministic data loader to avoid re-training on data upon resuming (eg if you do a total of <=1 epoch), so having an option for determinism could be great even beyond debugging use cases.

**Comment by kevinzakka** [comment]
@kpertsch I did some testing in a notebook and it seemed deterministic but have to write a full unit test.

**Comment by peter-mitrano-bg** [comment]
I would find this addition useful, although possible we should also expose the arg in the `from_tfrecords` function not just `from_rlds`?

**Comment by peter-mitrano-bg** [comment]
And of course I tested and can confirm this works. Having this hard-coded to false is extra scary because the TF docs say that if `shuffle_files=False` then it will automatically set `deterministic` to true, but that I guess doesn't happen in this case :skull: 

https://www.tensorflow.org/datasets/determinism#with_shuffle_filestrue
